### DISCUSSION TITLE
Tasmota flashed Smart Switch, Zigbee and Tasmota Flashed Sonoff ZigBee Bridge Pro

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18529] [Q&A] [answered]
I have a 16Amp Tasmota Flashed ESP8266 based Smart Switch. It has been flashed with Custom Tasmota Binaray generated through online Tasmo Compiler using ZigBee option. (1) What purpose does selecting ZigBee option during compile Serve? 

I have noticed that it adds following custom lines:

```
//Note: USE_ZIGBEE_ZNP and USE_ZIGBEE_EZSP are mutually incompatible, you must select exactly one

// Enable ZNP protocol, needed for CC2530 based devices
#define USE_ZIGBEE_ZNP
// Enable EZSP protocol, needed for EFR32 EmberZNet based devices, like Sonoff Zigbee bridge
//#define USE_ZIGBEE_EZSP
//#ifdef USE_ZIGBEE_ZNP
//  #undef USE_ZIGBEE_ZNP
//#endif
```

and on the flashed device it adds option for ZigBee RST, ZigBee TX and Zibee Rx under "Configure Template" menu. Assigning these options to GPIO gives following "Main Menu"screen with following buttons inactivated "ZigBee Permit Join" and "ZigBee Map". Please see the attached Picture.
![Screenshot (208)](https://user-images.githubusercontent.com/69470381/234861147-06834c5e-dd0f-4cab-962d-3c7307f29ff1.png)

Next two questions:

(2)I was hoping that this device would be able to become part of ZigBee Network, Since Both wifi and Zigbee use 2.4Ghz frequency. Am I just missing some code/settings or am I barking up the wrong tree, thinking I can convert a Tasmota flashed wifi device to a ZigBee Device and offload all devices from wifi to Zigbee?

(3) There is a separate discussion (https://github.com/arendst/Tasmota/discussions/17872) about possibility of Bridge Mode on Tasmota Flashed Sonoff ZigBee Bridge (ESP32 Based) That may make it possible to have Tasmota on ESP32 control other Tasmota devices based on ESP8266 on the same LAN. If this feature is indeed made available on a Tasmota Sonoff BridgePRO will it be a solution for what I am trying to accomplish? or will the Tasmota flashed devices still keep using Wifi, Through the home wifi router, to be controlled by the Sonoff BridgePRO?

I apologize if above questions don't make sense from Protocol point of view. I am still trying to understand how ZigBee, Fits in with Wifi and then there is "Matter Protocol" which I just read about and may hold the answer.

EDIT: I think the answer for question (1) Just dawned on me, that a ZigBee Chip needs to interface with GPIOs to enable the "ZigBee Permit Join" and "ZigBee Map" buttons. Hmmmm a future tinkering project!

**Comment by sfromis** [comment, marked as answer]
The key role of Zigbee needing a specialized chip is also the answer to your (2) and (3). What Tasmota supports with Zigbee is to act as a coordinator in tandem with a suitable chip flashed with coordinator firmware, along with the suitable Tasmota build. And each Zigbee network only has **one** coordinator. While you can flash router firmware on a Zigbee chip, that removes the coordinator function, and hence any involvement from Tasmota. And devices without a specialized Zigbee-capable chip cannot be part of a Zigbee network. The technical detail that the 2.4 GHz band is used for both Zigbee and Wifi does not matter to this. 

With Tasmota, you could make your own Zigbee/Wifi bridging, but such work would be up to you, as it is just completely separate networks.

**Reply by NsinghP** [reply]
Ok, Now it is making sense, but this means that when Sonoff ZigBee Bridge Pro is given the anticipated Bridge Mode feature then it will stop working as a coordinator, hence another coordinator will need to be commissioned if there are any ZigBee devices on the Network. Also am  correct in concluding that a Sonoff ZigBee Bridge Pro  in Tasmota Flashed Bridge Mode will become single point of contact between Home Wifi Router and all Tasmota Devices thereby removing the load from home wifi router network?

**Reply by sfromis** [reply]
If you waste the capabilities of the Sonoff Zigbee bridge by demoting it to be a Zigbee router, it will not be a single point of contact, it will just be yet another node in the Zigbee mesh network, like what you already typically get with mains-powered Zigbee devices like plugs or bulbs. It will not have the coordinator role enabling it to reach outside of the Zigbee mesh network.

Note that I'm choosing to use the Zigbee term "router", and deliberately avoiding "bridge" (which is not a Zigbee term), but you could say that the single coordinator is doing bridging to whatever network it is connected to (typically Wifi or Ethernet), or as part of a smarthome server. Unless you find that you need extra nodes to improve a Zigbee mesh, do not consider flashing router firmware to a device designed to be a coordinator. You can get Zigbee routers in already mentioned devices, or as a small separate device.

**Reply by NsinghP** [reply]
Understood. You have just dashed all my hopes of justifying purchase of this Sonoff ZigBee BridgePro, other than, of-course, pure joy of successfully flashing it with Tasmota. Now I will have to buy a cheap Zigbee device to justify this coordinator. :) , and again thanks for your time.
